src/dhtproto/client/NotifierTypes.d

```diff
@@ -89,3 +134,43 @@ public struct RequestKeyInfo
             this.request_id, this.key);
     }
 }
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for method to serialize a record value.
+
+    Params:
+        dst = pointer to the buffer to be serialized into
+
+    TODO: deprecated, replace with implementation in swarm
+
+*******************************************************************************/
+
+template SerializeMethod ( alias dst )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Serializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Serializes `src` into `dst`, using a version decorater, if required.
+
+        Params:
+            T = type of struct to serialize
+            src = instance to serialize
+
+    ***************************************************************************/
+
+    public void serialize ( T ) ( T src )
+    {
+        static if ( Version.Info!(T).exists )
+            VersionDecorator.store!(T)(src, *dst);
```

Comment:
BTW, with respect to the singleton `VersionDecorator` instance used by `swarm.neo.client.mixins.DeserializeMethod`: given that only `loadCopy` is used, it's not actually necessary for the singleton instance to be exposed directly.  It should suffice to have a `public` function that uses a `private VersionDecorator` instance.

Reply:
:+1: that would be a slight improvement.

Reply:
(A question for swarm, of course. Not related to this PR.)